src/planning.py

```diff
@@ -417,6 +418,79 @@ def _run_low_level_search(task: Task, option_model: _OptionModelBase,
     return [], True
 
 
+def _run_low_level_plan(
+        task: Task, option_model: _OptionModelBase, plan: List[_Option],
+        timeout: float, max_horizon: int
+) -> Tuple[LowLevelTrajectory, bool]:  # pragma: no cover
+    """Runs a plan on an option model to generate a low level trajectory.
+
+    Returns a LowLevelTrajectory and a boolean. If the boolean is True,
+    the option sequence successfully executed to achieve the goal and
+    generated a LowLevelTrajectory. Otherwise, it returns and empty list
+    and False.
+    """
+    start_time = time.time()
+    cur_idx = 0
+    # The number of actions taken by each option in the plan. This is to
+    # make sure that we do not exceed the task horizon.
+    num_actions_per_option = [0 for _ in plan]
+    traj: List[State] = [task.init] + [DefaultState for _ in plan]
+    actions: List[Action] = [Action(np.array([0.0])) for _ in plan]
+    while cur_idx < len(plan):
+        if time.time() - start_time > timeout:
+            return LowLevelTrajectory([], [], False), False
+        state = traj[cur_idx]
+        option = plan[cur_idx]
+        cur_idx += 1
+        if option.initiable(state):
+            try:
+                next_state, num_actions = \
+                    option_model.get_next_state_and_num_actions(state, option)
+                print('Success')
+            except EnvironmentFailure as e:
+                can_continue_on = False
+                return LowLevelTrajectory([], [], False), False
+            else:  # an EnvironmentFailure was not raised
+                num_actions_per_option[cur_idx - 1] = num_actions
+                traj[cur_idx] = next_state
+                action_option = ParameterizedOption(
+                    option.parent.name, option.parent.types,
+                    option.parent.params_space,
+                    lambda s, m, o, p: Action(np.array([0.0])),
+                    lambda s, m, o, p: False,
+                    lambda s, m, o, p: True).ground(option.objects,
+                                                    option.params)
+                action_option.name = option.name
+                action_option.memory = option.memory
+                actions[cur_idx - 1].set_option(action_option)
```

Comment:
why can't you just do `actions[cur_idx - 1].set_option(option)`? why do you need a different thing called `action_option`? (please add a comment explaining this if there's a good reason, since it's a bit confusing!)